docs/tutorial/Tutorial.md

```diff
@@ -0,0 +1,1064 @@
+
+          
+Next, we need to change the mutator, in `mutator.rs`, to allocate to the 
+tospace, and to the two spaces controlled by the common plan. 
+1. Change the following import statements:
+   1. Add `use super::MyGC;`.
+   2. Add `use crate::util::alloc::BumpAllocator;`.
+   3. Delete `use crate::plan::mygc::MyGC;`.
+   
+   [[Finished code (step 1)]](/docs/tutorial/tutorial%20code/mygc_semispace/mutator.rs#L1-L12)
+
+2. In `lazy_static!`, make the following changes to `ALLOCATOR_MAPPING`, 
+which maps the required allocation semantics to the corresponding allocators. 
+For example, for `Default`, we allocate using the first bump pointer allocator 
+(`BumpPointer(0)`):
+   1. Map `Default` to `BumpPointer(0)`.
+   2. Map `ReadOnly` to `BumpPointer(1)`.
+   3. Map `Los` to `LargeObject(0)`. 
+   
+   [[Finished code (step 2)]](/docs/tutorial/tutorial%20code/mygc_semispace/mutator.rs#L39-L46)
+   
+3. Next, in `create_mygc_mutator`, change which allocator is allocated to what 
+space in `space_mapping`. Note that the space allocation is formatted as a list 
+of tuples. For example, the first bump pointer allocator (`BumpPointer(0)`) is 
+bound with `tospace`. 
+   1. `BumpPointer(0)` should map to the tospace.
+   2. `BumpPointer(1)` should map to `plan.common.get_immortal()`.
+   3. `LargeObject(0)` should map to `plan.common.get_los()`.
+   4. None of the above should be dereferenced (ie, they should not have 
+   the `&` prefix).
+   
+   [[Finished code (step 3)]](/docs/tutorial/tutorial%20code/mygc_semispace/mutator.rs#L48-L65)
+     
+There may seem to be 2 extraneous spaces and allocators that have appeared all 
+of a sudden in these past 2 steps. These are parts of the MMTk common plan 
+itself.
+ 1. The immortal space is used for objects that the virtual machine or a 
+ library never expects to die.
+ 2. The large object space is needed because MMTk handles particularly large 
+ objects differently to normal objects, as the space overhead of copying 
+ large objects is very high. Instead, this space is used by a free list 
+ allocator in the common plan to avoid having to copy them. 
+
+With this, you should have the allocation working, but not garbage collection. 
+Try building MyGC now. If you run HelloWorld or Fannkunchredux, they should 
+work. DaCapo's lusearch should fail, as it requires garbage to be collected. 
+   
+   
+   
+### Collection: Implement garbage collection
+
+#### CopyContext and Scheduler
+
+We need to add a few more things to get garbage collection working. 
+Specifically, we need to add a `CopyContext`, which a GC worker uses for 
+copying objects, and GC work packets that will be scheduled for a collection.
+
+At the moment, none of the files in the plan are suited for garbage collection 
+operations. So, we need to add a new file to hold the `CopyContext` and other 
+structures and functions that will give the collector proper functionality.
+
+1. Make a new file under `mygc`, called `gc_works.rs`. 
```

Comment:
Systemic:  this should be `gcwork` not `gcworks` (see #232)

Reply:
Just to confirm - this was the one you said I should ignore for now, since it's a systemic issue?